src/sqlfluff/dialects/dialect_exasol.py

```diff
@@ -872,8 +880,15 @@ class TableOutOfLineConstraintSegment(BaseSegment):
     """Out of line table constraint for CREATE / ALTER TABLE."""
 
     type = "table_constraint_definition"
-    match_grammar = Sequence(
-        Sequence("CONSTRAINT", Ref("SingleIdentifierGrammar"), optional=True),
+    match_grammar = StartsWith(
+        OneOf("CONSTRAINT", "PRIMARY", "FOREIGN"),
+    )
+    parse_grammar = Sequence(
+        Sequence(
+            "CONSTRAINT",
+            Ref("NakedIdentifierSegment", optional=True),
+            optional=True,
+        ),
```

Comment:
Same here:

```suggestion
    parse_grammar = Sequence(
        Sequence(
            "CONSTRAINT",
            AnyNumberOf(
                Ref("NakedIdentifierSegment"),
                max_times=1,
                min_times=0,
                exclude=OneOf("PRIMARY", "FOREIGN"),
            ),
            optional=True,
        ),
```

Reply:
Thank you so much. It needed some extra adjustment but worked quite well 🚀